# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/nsIQuotaManagerService.idl
# Commit: d63dac1ee4d4
# Full Hash: d63dac1ee4d471defff484dae5a3535e1f8ba072
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-10 22:03:45
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Make it possible to initialize temporary origins without ensuring origin directories; r=dom-storage-reviewers,jari
#   
#   LSNG already uses some QuotaManager APIs to achieve that origin directories are
#   not created if they don't exist during datastore preparation, but the feature
#   is not easy to use and it's also not generalized enough for use in other quota
# ==============================================================================

diff -r 2ab08c49f2b1 -r d63dac1ee4d4 dom/quota/nsIQuotaManagerService.idl
--- a/dom/quota/nsIQuotaManagerService.idl	Thu Oct 10 12:22:54 2024 +0000
+++ b/dom/quota/nsIQuotaManagerService.idl	Thu Oct 10 12:22:55 2024 +0000
@@ -116,10 +116,15 @@
    *
    * @param aPrincipal
    *        A principal for the origin whose directory is to be initialized.
+   *
+   * @param aCreateIfNonExistent
+   *        An optional boolean to indicate creation of origin directory if it
+   *        doesn't exist yet.
    */
   [must_use] nsIQuotaRequest
   initializeTemporaryOrigin(in ACString aPersistenceType,
-                            in nsIPrincipal aPrincipal);
+                            in nsIPrincipal aPrincipal,
+                            [optional] in boolean aCreateIfNonExistent);
 
   /**
    * Initializes persistent client directory for the given origin and client.